internal/provider/provider.go

```diff
@@ -37,6 +49,33 @@ func (p *Provider) Configure(ctx context.Context, req provider.ConfigureRequest,
 	if resp.Diagnostics.HasError() {
 		return
 	}
+
+	kc := authn.NewMultiKeychain(google.Keychain, authn.DefaultKeychain)
+	ropts := []remote.Option{
+		remote.WithAuthFromKeychain(kc),
+		remote.WithUserAgent("terraform-provider-apko/" + p.version),
+	}
+
+	puller, err := remote.NewPuller(ropts...)
+	if err != nil {
+		resp.Diagnostics.AddError("Configure []remote.Option", err.Error())
```

Comment:
These are surfaced to users, so we should make them less sausage-y

Reply:
Like so? FWIW, these errors can't happen as the code is currently written, I'm just checking them out of paranoia.